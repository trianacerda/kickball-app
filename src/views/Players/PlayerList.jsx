import { useEffect, useState } from 'react';
import { NavLink, Link } from 'react-router-dom';
import { getPlayers, deletePlayerById } from '../../services/players';
import AddPlayer from './AddPlayer';

function PlayerList() {
  const [loading, setLoading] = useState(true);
  const [players, setPlayers] = useState([]);

  const loadPlayers = async () => {
    setLoading(true);
    const resp = await getPlayers();
    setPlayers(resp);
    setLoading(false);
  };
  useEffect(() => {
    loadPlayers();
  }, []);

  const handleDelete = async ({ id }) => {
    const shouldDelete = confirm(`Are you sure you wanna delete ${name}??`);

    if (shouldDelete) {
      await deletePlayerById(id);
      await loadPlayers();
    }
  };

  if (loading) return <h1>Loading players.....</h1>;

  return (
    <>
      <NavLink
        to="/"
        className="home-link"
        exact
        style={{ display: 'flex', flexDirection: 'start' }}
      >
        Home
      </NavLink>
      <NavLink
        to="/teams"
        className="team-link"
        exact
        style={{ display: 'flex', flexDirection: 'start' }}
      >
        Teams
      </NavLink>
      <h1>List of Players:</h1>
      <NavLink
        to="/players/new"
        className="new-player-link"
        exact
        style={{ display: 'flex', flexDirection: 'column-reverse' }}
      >
        Add a Player
      </NavLink>
      <ul aria-label="players-list">
        {players.map((player) => {
          return (
            <div key={player.id} style={{ border: 'solid', borderColor: 'lightseagreen' }}>
              <li key={player.id}>
                <Link to={`/players/${player.id}`} className="player-list-link">
                  {player.name}
                </Link>
                <NavLink
                  to={`/players/${player.id}/update`}
                  className="update-player-link"
                  exact
                  style={{ display: 'flex', flexDirection: 'column-reverse', margin: '15px' }}
                >
                  Update Player
                </NavLink>
                <button
                  aria-label={`Delete${player.name}`}
                  type="button"
                  value="delete"
                  onClick={() => {
                    handleDelete({ id: player.id, name: player.name });
                  }}
                >
                  Delete
                </button>
              </li>
            </div>
          );
        })}
      </ul>
    </>
  );
}

export default PlayerList;
