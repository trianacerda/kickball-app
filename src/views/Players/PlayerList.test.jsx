import { render, screen } from '@testing-library/react';
import { MemoryRouter, Route } from 'react-router-dom';
import PlayerList from './PlayerList';

it('should render a list of players', async () => {
  const { container } = render(
    <MemoryRouter>
      <PlayerList />
    </MemoryRouter>
  );

  const team1 = await screen.findByText('Bella Boo');
  const team2 = await screen.findByText('Jonny Clutch');
  const playerLabel = await screen.findByRole('list', { name: 'players-list' });

  expect(team1).toBeInTheDocument();
  expect(team2).toBeInTheDocument();
  expect(playerLabel).toBeInTheDocument();
  expect(container).toMatchSnapshot();
});
