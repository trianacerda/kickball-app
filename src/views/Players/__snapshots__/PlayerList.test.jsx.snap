// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render a list of players 1`] = `
<div>
  <a
    aria-current="page"
    class="home-link active"
    href="/"
    style="display: flex; flex-direction: start;"
  >
    Home
  </a>
  <a
    class="team-link"
    href="/teams"
    style="display: flex; flex-direction: start;"
  >
    Teams
  </a>
  <h1>
    List of Players:
  </h1>
  <a
    class="new-player-link"
    href="/players/new"
    style="display: flex; flex-direction: column-reverse;"
  >
    Add a Player
  </a>
  <ul
    aria-label="players-list"
  >
    <div
      style="border: solid; border-color: lightseagreen;"
    >
      <li>
        <a
          class="player-list-link"
          href="/players/29"
        >
          134
        </a>
        <a
          class="update-player-link"
          href="/players/29/update"
          style="display: flex; flex-direction: column-reverse; margin: 15px;"
        >
          Update Player
        </a>
        <button
          aria-label="Delete134"
          type="button"
          value="delete"
        >
          Delete
        </button>
      </li>
    </div>
    <div
      style="border: solid; border-color: lightseagreen;"
    >
      <li>
        <a
          class="player-list-link"
          href="/players/17"
        >
          Ace Auburn
        </a>
        <a
          class="update-player-link"
          href="/players/17/update"
          style="display: flex; flex-direction: column-reverse; margin: 15px;"
        >
          Update Player
        </a>
        <button
          aria-label="DeleteAce Auburn"
          type="button"
          value="delete"
        >
          Delete
        </button>
      </li>
    </div>
    <div
      style="border: solid; border-color: lightseagreen;"
    >
      <li>
        <a
          class="player-list-link"
          href="/players/18"
        >
          Aven Garrett
        </a>
        <a
          class="update-player-link"
          href="/players/18/update"
          style="display: flex; flex-direction: column-reverse; margin: 15px;"
        >
          Update Player
        </a>
        <button
          aria-label="DeleteAven Garrett"
          type="button"
          value="delete"
        >
          Delete
        </button>
      </li>
    </div>
    <div
      style="border: solid; border-color: lightseagreen;"
    >
      <li>
        <a
          class="player-list-link"
          href="/players/2"
        >
          Bella Boo
        </a>
        <a
          class="update-player-link"
          href="/players/2/update"
          style="display: flex; flex-direction: column-reverse; margin: 15px;"
        >
          Update Player
        </a>
        <button
          aria-label="DeleteBella Boo"
          type="button"
          value="delete"
        >
          Delete
        </button>
      </li>
    </div>
    <div
      style="border: solid; border-color: lightseagreen;"
    >
      <li>
        <a
          class="player-list-link"
          href="/players/16"
        >
          Calton Bryden
        </a>
        <a
          class="update-player-link"
          href="/players/16/update"
          style="display: flex; flex-direction: column-reverse; margin: 15px;"
        >
          Update Player
        </a>
        <button
          aria-label="DeleteCalton Bryden"
          type="button"
          value="delete"
        >
          Delete
        </button>
      </li>
    </div>
    <div
      style="border: solid; border-color: lightseagreen;"
    >
      <li>
        <a
          class="player-list-link"
          href="/players/4"
        >
          Charleston Dax
        </a>
        <a
          class="update-player-link"
          href="/players/4/update"
          style="display: flex; flex-direction: column-reverse; margin: 15px;"
        >
          Update Player
        </a>
        <button
          aria-label="DeleteCharleston Dax"
          type="button"
          value="delete"
        >
          Delete
        </button>
      </li>
    </div>
    <div
      style="border: solid; border-color: lightseagreen;"
    >
      <li>
        <a
          class="player-list-link"
          href="/players/5"
        >
          Crispin Cruz
        </a>
        <a
          class="update-player-link"
          href="/players/5/update"
          style="display: flex; flex-direction: column-reverse; margin: 15px;"
        >
          Update Player
        </a>
        <button
          aria-label="DeleteCrispin Cruz"
          type="button"
          value="delete"
        >
          Delete
        </button>
      </li>
    </div>
    <div
      style="border: solid; border-color: lightseagreen;"
    >
      <li>
        <a
          class="player-list-link"
          href="/players/6"
        >
          Declan Ebenezer
        </a>
        <a
          class="update-player-link"
          href="/players/6/update"
          style="display: flex; flex-direction: column-reverse; margin: 15px;"
        >
          Update Player
        </a>
        <button
          aria-label="DeleteDeclan Ebenezer"
          type="button"
          value="delete"
        >
          Delete
        </button>
      </li>
    </div>
    <div
      style="border: solid; border-color: lightseagreen;"
    >
      <li>
        <a
          class="player-list-link"
          href="/players/7"
        >
          Ike Jaspin
        </a>
        <a
          class="update-player-link"
          href="/players/7/update"
          style="display: flex; flex-direction: column-reverse; margin: 15px;"
        >
          Update Player
        </a>
        <button
          aria-label="DeleteIke Jaspin"
          type="button"
          value="delete"
        >
          Delete
        </button>
      </li>
    </div>
    <div
      style="border: solid; border-color: lightseagreen;"
    >
      <li>
        <a
          class="player-list-link"
          href="/players/1"
        >
          Jonny  
Clutch
        </a>
        <a
          class="update-player-link"
          href="/players/1/update"
          style="display: flex; flex-direction: column-reverse; margin: 15px;"
        >
          Update Player
        </a>
        <button
          aria-label="DeleteJonny  
Clutch"
          type="button"
          value="delete"
        >
          Delete
        </button>
      </li>
    </div>
    <div
      style="border: solid; border-color: lightseagreen;"
    >
      <li>
        <a
          class="player-list-link"
          href="/players/8"
        >
          Kai Judson
        </a>
        <a
          class="update-player-link"
          href="/players/8/update"
          style="display: flex; flex-direction: column-reverse; margin: 15px;"
        >
          Update Player
        </a>
        <button
          aria-label="DeleteKai Judson"
          type="button"
          value="delete"
        >
          Delete
        </button>
      </li>
    </div>
    <div
      style="border: solid; border-color: lightseagreen;"
    >
      <li>
        <a
          class="player-list-link"
          href="/players/9"
        >
          Kayden Kale
        </a>
        <a
          class="update-player-link"
          href="/players/9/update"
          style="display: flex; flex-direction: column-reverse; margin: 15px;"
        >
          Update Player
        </a>
        <button
          aria-label="DeleteKayden Kale"
          type="button"
          value="delete"
        >
          Delete
        </button>
      </li>
    </div>
    <div
      style="border: solid; border-color: lightseagreen;"
    >
      <li>
        <a
          class="player-list-link"
          href="/players/19"
        >
          Kingston Lathan
        </a>
        <a
          class="update-player-link"
          href="/players/19/update"
          style="display: flex; flex-direction: column-reverse; margin: 15px;"
        >
          Update Player
        </a>
        <button
          aria-label="DeleteKingston Lathan"
          type="button"
          value="delete"
        >
          Delete
        </button>
      </li>
    </div>
    <div
      style="border: solid; border-color: lightseagreen;"
    >
      <li>
        <a
          class="player-list-link"
          href="/players/20"
        >
          Leland Kole
        </a>
        <a
          class="update-player-link"
          href="/players/20/update"
          style="display: flex; flex-direction: column-reverse; margin: 15px;"
        >
          Update Player
        </a>
        <button
          aria-label="DeleteLeland Kole"
          type="button"
          value="delete"
        >
          Delete
        </button>
      </li>
    </div>
    <div
      style="border: solid; border-color: lightseagreen;"
    >
      <li>
        <a
          class="player-list-link"
          href="/players/15"
        >
          Marvel Oz
        </a>
        <a
          class="update-player-link"
          href="/players/15/update"
          style="display: flex; flex-direction: column-reverse; margin: 15px;"
        >
          Update Player
        </a>
        <button
          aria-label="DeleteMarvel Oz"
          type="button"
          value="delete"
        >
          Delete
        </button>
      </li>
    </div>
    <div
      style="border: solid; border-color: lightseagreen;"
    >
      <li>
        <a
          class="player-list-link"
          href="/players/3"
        >
          Mista Mista
        </a>
        <a
          class="update-player-link"
          href="/players/3/update"
          style="display: flex; flex-direction: column-reverse; margin: 15px;"
        >
          Update Player
        </a>
        <button
          aria-label="DeleteMista Mista"
          type="button"
          value="delete"
        >
          Delete
        </button>
      </li>
    </div>
    <div
      style="border: solid; border-color: lightseagreen;"
    >
      <li>
        <a
          class="player-list-link"
          href="/players/13"
        >
          Nellary Nestor
        </a>
        <a
          class="update-player-link"
          href="/players/13/update"
          style="display: flex; flex-direction: column-reverse; margin: 15px;"
        >
          Update Player
        </a>
        <button
          aria-label="DeleteNellary Nestor"
          type="button"
          value="delete"
        >
          Delete
        </button>
      </li>
    </div>
    <div
      style="border: solid; border-color: lightseagreen;"
    >
      <li>
        <a
          class="player-list-link"
          href="/players/14"
        >
          Ode Mountain
        </a>
        <a
          class="update-player-link"
          href="/players/14/update"
          style="display: flex; flex-direction: column-reverse; margin: 15px;"
        >
          Update Player
        </a>
        <button
          aria-label="DeleteOde Mountain"
          type="button"
          value="delete"
        >
          Delete
        </button>
      </li>
    </div>
    <div
      style="border: solid; border-color: lightseagreen;"
    >
      <li>
        <a
          class="player-list-link"
          href="/players/10"
        >
          Slash Taden
        </a>
        <a
          class="update-player-link"
          href="/players/10/update"
          style="display: flex; flex-direction: column-reverse; margin: 15px;"
        >
          Update Player
        </a>
        <button
          aria-label="DeleteSlash Taden"
          type="button"
          value="delete"
        >
          Delete
        </button>
      </li>
    </div>
    <div
      style="border: solid; border-color: lightseagreen;"
    >
      <li>
        <a
          class="player-list-link"
          href="/players/11"
        >
          Tevin Zee
        </a>
        <a
          class="update-player-link"
          href="/players/11/update"
          style="display: flex; flex-direction: column-reverse; margin: 15px;"
        >
          Update Player
        </a>
        <button
          aria-label="DeleteTevin Zee"
          type="button"
          value="delete"
        >
          Delete
        </button>
      </li>
    </div>
    <div
      style="border: solid; border-color: lightseagreen;"
    >
      <li>
        <a
          class="player-list-link"
          href="/players/31"
        >
          blast
        </a>
        <a
          class="update-player-link"
          href="/players/31/update"
          style="display: flex; flex-direction: column-reverse; margin: 15px;"
        >
          Update Player
        </a>
        <button
          aria-label="Deleteblast"
          type="button"
          value="delete"
        >
          Delete
        </button>
      </li>
    </div>
    <div
      style="border: solid; border-color: lightseagreen;"
    >
      <li>
        <a
          class="player-list-link"
          href="/players/12"
        >
          jon
        </a>
        <a
          class="update-player-link"
          href="/players/12/update"
          style="display: flex; flex-direction: column-reverse; margin: 15px;"
        >
          Update Player
        </a>
        <button
          aria-label="Deletejon"
          type="button"
          value="delete"
        >
          Delete
        </button>
      </li>
    </div>
    <div
      style="border: solid; border-color: lightseagreen;"
    >
      <li>
        <a
          class="player-list-link"
          href="/players/30"
        >
          tr
        </a>
        <a
          class="update-player-link"
          href="/players/30/update"
          style="display: flex; flex-direction: column-reverse; margin: 15px;"
        >
          Update Player
        </a>
        <button
          aria-label="Deletetr"
          type="button"
          value="delete"
        >
          Delete
        </button>
      </li>
    </div>
    <div
      style="border: solid; border-color: lightseagreen;"
    >
      <li>
        <a
          class="player-list-link"
          href="/players/28"
        >
          triana
        </a>
        <a
          class="update-player-link"
          href="/players/28/update"
          style="display: flex; flex-direction: column-reverse; margin: 15px;"
        >
          Update Player
        </a>
        <button
          aria-label="Deletetriana"
          type="button"
          value="delete"
        >
          Delete
        </button>
      </li>
    </div>
  </ul>
</div>
`;
